main.go

```diff
@@ -450,6 +539,67 @@ func (c *CASProxy) LookupJobStatusUpdates(w http.ResponseWriter, r *http.Request
 	fmt.Fprintln(w, string(js))
 }
 
+// URLIsReady returns true if the Ingress for the provided URL exists and if a
+// connection attempt to the Endpoint for the Ingress succeeds.
+func (c *CASProxy) URLIsReady(w http.ResponseWriter, r *http.Request) {
+	u := r.FormValue("url")
+
+	matches, err := c.URLMatches(u)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error checking URL %s: %s", u, err.Error()), http.StatusInternalServerError)
+		return
+	}
+	if !matches {
+		http.Error(w, fmt.Sprintf("URL %s is not a subdomain of %s", u, c.viceDomain), http.StatusBadRequest)
+		return
+	}
+
+	subdomain, err := extractSubdomain(u)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error getting subdomain for URL %s", u), http.StatusInternalServerError)
+		return
+	}
+	if subdomain == "" {
+		http.Error(w, fmt.Sprintf("empty subdomain for URL '%s'", u), http.StatusBadRequest)
+		return
+	}
+
+	var ready bool
+	ready, err = c.IngressExists(subdomain)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error checking ingress %s existence: %s", subdomain, err.Error()), http.StatusInternalServerError)
+		return
+	}
+
+	if ready {
+		var ept *Endpoint
+		ept, err = c.EndpointConfig(subdomain)
+		if err != nil {
+			log.Error(err)
+			ready = ready && false
+		}
+
+		var conn net.Conn
+		conn, err = net.Dial("tcp", fmt.Sprintf("%s:%d", ept.IP, ept.Port))
```

Comment:
I *think* this will be sufficient for a check to see if the app is responding, but I'm not 100% sure on that.

Reply:
I toyed with the idea of issuing an HTTP request, but at some point in the relatively near future we're going to have to support more than HTTP(S). That's why I decided to leave out the test request.